jaissica - added donut chart for hgn phase 2 dashboard

jaissica opened the pull request:
# Description
- Implemented donut Chart to show Proportions of Tools Available. 
- 
<img width="702" alt="Screenshot 2025-05-17 at 3 07 35 PM" src="https://github.com/user-attachments/assets/4a5cbbc2-db9e-4a20-8f66-04438c1f45ee" />



## Related PRS (if any):
Related backend PR: https://github.com/OneCommunityGlobal/HGNRest/pull/1331

## Main changes explained:
- Added styling in WeeklyProjectSummary.css
- Added donut chart in ToolStatusDonutChart.jsx
- Incorporated donut chart in the main dashboard WeeklyProjectSummary.jsx. 

## How to test:
1. check into current branch
2. do npm install and ... to run this PR locally
3. Clear site data/cache
4. log as an admin user
5. Navigate to: http://localhost:3000/bmdashboard/totalconstructionsummary
6. In the Project Status you would be able to see different buttons created.

## Screenshots or videos of changes:
<img width="476" alt="Screenshot 2025-03-22 at 1 21 28 AM" src="https://github.com/user-attachments/assets/7077662e-03aa-4fa6-b3d2-2c973a12447a" />


## Note:
You might see the styling to be a bit off on smaller viewports. I am awaiting changes related to Lessons Learner and Utilization Rate and Downtime of Tools/Equipment to better gauge the spacing and padding related changes.

angad363 reviewed (changes requested):
I followed the steps mentioned and tested the changes locally. The donut chart is rendering correctly and displays the proportions of tools available as expected. However, I found a few issues:

1. In dark mode, the font color of the text still appears black. It should be white for better visibility, unless this was intentionally excluded from the scope of this PR. Please confirm if this is something that should be fixed here or in a separate task.

2. There are several code quality issues flagged by GitHub actions, causing unit tests to fail. It would be beneficial to address these to ensure the project maintains its quality standards.

3. The component is responsive; however, in extreme small screen sizes, the chart overlaps neighboring components. I understand that the requester is awaiting another PR to resolve this, so it might be fixed in that upcoming change.

Please make the necessary code changes to address the issues with dark mode compatibility and fix the code quality problems. I've attached screenshots and a video for reference. Let me know how you'd like to proceed.

![PR #3302 - 1](https://github.com/user-attachments/assets/552b4a1c-388f-4699-b5cd-3298193985c8)
![PR #3302 - 2](https://github.com/user-attachments/assets/d880bd04-e2cf-4f0b-8733-f3645a2efca9)

https://github.com/user-attachments/assets/a5b8e0b7-c041-4942-b678-94778d389af0

Amalesh-A reviewed (changes requested):
I followed your instructions and checked out the jaissica-create-donut-chart-for-tools-available branch, installed the dependencies, and launched the app locally. I observed that it functions as expected:
![It works](https://github.com/user-attachments/assets/39259857-e3c1-4891-a2f4-25edf0903a67)

Also, just a suggestion — consider changing the font color to white when using dark mode:
![Dark](https://github.com/user-attachments/assets/6365f89f-2182-43bf-9f3c-a759d008697e)

jaissica commented:
> I followed the steps mentioned and tested the changes locally. The donut chart is rendering correctly and displays the proportions of tools available as expected. However, I found a few issues:
> 
> 1. In dark mode, the font color of the text still appears black. It should be white for better visibility, unless this was intentionally excluded from the scope of this PR. Please confirm if this is something that should be fixed here or in a separate task.
> 2. There are several code quality issues flagged by GitHub actions, causing unit tests to fail. It would be beneficial to address these to ensure the project maintains its quality standards.
> 3. The component is responsive; however, in extreme small screen sizes, the chart overlaps neighboring components. I understand that the requester is awaiting another PR to resolve this, so it might be fixed in that upcoming change.
> 
> Please make the necessary code changes to address the issues with dark mode compatibility and fix the code quality problems. I've attached screenshots and a video for reference. Let me know how you'd like to proceed.
> 
> ![PR #3302 - 1](https://private-user-images.githubusercontent.com/37550965/425752476-552b4a1c-388f-4699-b5cd-3298193985c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3MDgwNzUsIm5iZiI6MTc0MzcwNzc3NSwicGF0aCI6Ii8zNzU1MDk2NS80MjU3NTI0NzYtNTUyYjRhMWMtMzg4Zi00Njk5LWI1Y2QtMzI5ODE5Mzk4NWM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDAzVDE5MTYxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxYzQyMGViMzY4NDc4ZGY5M2U2MzE5M2NlYjM2MzQ5OGYxOTk2YTU0YzFhNDZmMjRjODVjZjFkMWY3MTNmODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y5YP0_zneu1k1TbVJAHuTGBYqzCWSfZ9RvwaUu1IAZE) ![PR #3302 - 2](https://private-user-images.githubusercontent.com/37550965/425752477-d880bd04-e2cf-4f0b-8733-f3645a2efca9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3MDgwNzUsIm5iZiI6MTc0MzcwNzc3NSwicGF0aCI6Ii8zNzU1MDk2NS80MjU3NTI0NzctZDg4MGJkMDQtZTJjZi00ZjBiLTg3MzMtZjM2NDVhMmVmY2E5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDAzVDE5MTYxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyODNjODRiMTA2MTkyMGE5MzY5OTlhN2IyOGZmMmFjYzAyNTJlNjcyZDRkZTk2MzI0NmVjNWMzMWNjMDI5NjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.a59AQ77-KhPd1h7tLSWn--5feZuFcX4LpI3b8LrHbps)
> 
>  PR.3301.-.3.mp4

Thanks for the feedback! I have addressed the changes mentioned in (1) & (2). For (2) I have fixed issues related to linting and the failing test(`npm run test`) seems to be independent of changes in the PR.

nikhilpittala16 reviewed (dismissed):
I have tested the PR and everything works as expected.
![image](https://github.com/user-attachments/assets/502aaeff-30c1-41a8-99b1-d6f89ed60aff)

Ram-blip reviewed (commented):
I followed all the steps, but when I tried to review the PR, the corresponding page didn’t appear.

<img width="1470" alt="PR 3301" src="https://github.com/user-attachments/assets/fa6d6f2d-e24b-413b-8ca0-0fe02fb345cb" />

TANMAY-AR0RA reviewed (changes requested):
Initially, the page was blank with no data. There was no mention of a corresponding backend PR, but I found one with the same name, and the chart loaded after that.

The page is responsive. In dark mode, the labels "Tool/Equipment Name" and "Project" are hard to read. 

<b> I suggest changing the text color to white. I've highlighted them with red rectangles.


![image](https://github.com/user-attachments/assets/210cc9e8-719c-4980-a294-c57b73c0be98)

jaissica commented:
> Initially, the page was blank with no data. There was no mention of a corresponding backend PR, but I found one with the same name, and the chart loaded after that.
> 
> The page is responsive. In dark mode, the labels "Tool/Equipment Name" and "Project" are hard to read.
> 
> I suggest changing the text color to white. I've highlighted them with red rectangles.
> 
> ![image](https://private-user-images.githubusercontent.com/107872653/462828567-210cc9e8-719c-4980-a294-c57b73c0be98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIwMjYyNzIsIm5iZiI6MTc1MjAyNTk3MiwicGF0aCI6Ii8xMDc4NzI2NTMvNDYyODI4NTY3LTIxMGNjOWU4LTcxOWMtNDk4MC1hMjk0LWM1N2I3M2MwYmU5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcwOVQwMTUyNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YjhkNzI1NmNlMzBkMDNkMTRiZDJhZDQ3YWVlNTMwN2NmODFmOTM1OGFiMmJjODM3MGMyZTM1NmY0ZTNiYWMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6S_8Ha2RSPdQFv4SK3p1YHvVySZoXjxViGUiJb6Ex44)

Thanks for the feedback. I’ve updated the backend PR here: https://github.com/OneCommunityGlobal/HGNRest/pull/1331.
Please review both the frontend and backend changes together — you should now be able to see the records as expected.
Additionally, I’ve adjusted the dark mode colors to improve readability.

Babu-kota reviewed (approved):
The visual representation of tool availability is a solid enhancement to the dashboard 
clean integration in Weekly Project Summary. and the modular approach with Tool Status Donut Chart is well-structured.

<img width="795" height="962" alt="jessicia" src="https://github.com/user-attachments/assets/71f0c663-8c28-4b1b-b829-79eafb57238f" />
